core/core-configuration-layer.el

```diff
@@ -119,11 +119,28 @@ sub-directory of the contribution directory.")
   (when dotspacemacs-delete-orphan-packages
     (configuration-layer/delete-orphan-packages)))
 
-(defun configuration-layer/create-layer (name)
-  "Ask the user for a configuration layer name and create a layer with this
-name in the private layers directory."
-  (interactive "sConfiguration layer name: ")
-  (let ((layer-dir (configuration-layer//get-private-layer-dir name)))
+(defun configuration-layer//create-layer-source ()
+  "return a source for helm selection of layer directory"
+  `((name . "Current Configuration Layer Paths")
+    (candidates . ,(append (mapcar (lambda (dir) (expand-file-name dir))
+                                   (cl-pushnew configuration-layer-private-directory
+                                               dotspacemacs-configuration-layer-path))
+                           (list "*Create new layer path*")))
+    (action . (lambda (candidate) candidate))))
+
+(defun configuration-layer/create-layer ()
+  "Ask the user for a configuration layer name and the layer
+directory to use. Create a layer with this name in the selected
+layer directory."
+  (interactive)
+  (let* ((layer-path-sel (helm
+                          :sources (list (configuration-layer//create-layer-source))
+                          :prompt "Configuration Layer Path: "))
+         (layer-path (if (not (file-exists-p layer-path-sel))
+                         (read-file-name "New layer path (new or existing directory): " "~/" )
+                       layer-path-sel))
```

Comment:
To be more explicit you can add an entry to the sources (line 137) called `Another directory...` and test `layer-path-sel` against this value.
If `layer-path-sel` is neither of the provided candidates then do nothing and warn about it.